### DISCUSSION TITLE
how to create oracle cdc table?

### DISCUSSION THREAD

**Discussion by A-little-bit-of-data** [apache/flink-cdc#533] [Q&A]
CREATE TABLE oracle_binlog (
 id INT NOT NULL,
 name STRING,
 description STRING,
 weight DECIMAL(10,3)
) WITH (
 'connector' = 'oracle-cdc',
 'hostname' = '',
 'port' = '',
 'username' = '',
 'password' = '',
 'database-name' = '',
 'table-name' = ''
);
This is for reference to create a MySQL CDC table, but this does not work properly.

**Comment by leonardBang** [comment]
@luoyuxia contribution `oracle-cdc` documentation, maybe you can have a look
https://ververica.github.io/flink-cdc-connectors/master/content/connectors/oracle-cdc.html

**Reply by A-little-bit-of-data** [reply]
> @luoyuxia contribution `oracle-cdc` documentation, maybe you can have a look https://ververica.github.io/flink-cdc-connectors/master/content/connectors/oracle-cdc.html


> @luoyuxia contribution `oracle-cdc` documentation, maybe you can have a look https://ververica.github.io/flink-cdc-connectors/master/content/connectors/oracle-cdc.html

Flink SQL> CREATE TABLE products (
     id INT NOT NULL,
     name STRING,
     description STRING,
     weight DECIMAL(10, 3),
     PRIMARY KEY(id) NOT ENFORCED
     ) WITH (
     'connector' = 'oracle-cdc',
     'hostname' = 'localhost',
     'port' = '1521',
     'username' = 'flinkuser',
     'password' = 'flinkpw',
     'database-name' = 'XE',
     'schema-name' = 'inventory',
     'table-name' = 'products');

What are the database-name and schema-name here? Are all the library names of the database?

**Reply by luoyuxia** [reply]
@A-little-bit-of-data 
Hi, 
`database-name` is the intance name of your oracle, usually known as SID.
As for `schema-name`,  
when you use such ddl `create table test. test_table ` to create table,  then the `test` is the `schema-name`; 
 when you use such ddl `create table test_table`  without prefix to create table,  then the `schema-name` is usually the `username`  you used to login to oracle .

Hopes it can help, feel free to ask if you have any other question.

**Reply by A-little-bit-of-data** [reply]
> @A-little-bit-of-data Hi, `database-name` is the intance name of your oracle, usually known as SID. As for `schema-name`, when you use such ddl `create table test. test_table ` to create table, then the `test` is the `schema-name`; when you use such ddl `create table test_table` without prefix to create table, then the `schema-name` is usually the `username` you used to login to oracle .
> 
> Hopes it can help, feel free to ask if you have any other question.

Hello,
1. I placed flink-connector-oracle-cdc-2.1-SNAPSHOT.jar in the <FLINK_HOME>/lib/ directory of each machine in the cluster, and started the sql-client cluster in standalone mode. This package was not loaded and an error was reported: [ERROR] Could not execute SQL statement. Reason:
java.lang.ClassNotFoundException: com.ververica.cdc.debezium.DebeziumSourceFunction

2. When I change the startup command to ./sql-client.sh embedded -i /usr/local/sql-client.sql -j /usr/local/flink-1.13.2/lib/flink-sql-connector- When oracle-cdc-2.1-SNAPSHOT.jar is created, the following error is reported when the table query is completed: [ERROR] Could not execute SQL statement. Reason:
java.lang.AbstractMethodError: Method org/apache/kafka/connect/json/JsonSerializer.configure(Ljava/util/Map;Z)V is abstract
Please answer when you have time, thank you!

**Reply by Rowen110** [reply]
你的问题 解决方案：
问题1: 需要注释掉你flink/lib下的 mysql cdc jar 目前它和oracle cdc jar 会有冲突，导致异常 会报“ java.lang.ClassNotFoundException: com.ververica.cdc.debezium.DebeziumSourceFunction” 这个异常
问题2: 我理解还是需要 oracle cdc jar 在启动flink集群服务前，copy到 flink_home lib下 再拉起集群服务，并保障它加载成功有效。具体可看jobmanager log详情
希望能帮到你～

**Comment by luoyuxia** [comment]
Where do you get from `flink-connector-oracle-cdc-2.1-SNAPSHOT.jar`. It seems there are some depedencies miss.

Maybe you can build the module `flink-sql-connector-oracle-cdc` .

And put it into <FLINK_HOME>/lib/ , then run with ` sql-client`. Hopes it can help.

**Reply by A-little-bit-of-data** [reply]
I compiled the source code package downloaded from the official website by myself, followed the steps of the official website, and finally showed that the compilation was successful.

**Reply by A-little-bit-of-data** [reply]
![image](https://user-images.githubusercontent.com/31812278/139370285-bd511f66-d8da-4b57-95b5-8e43ac143d2d.png)
The URL in this interface is 404.
This URL: https://mvnrepository.com/search?q=com.ververica&d=com.ververica cannot find flink-connector-oracle-cdc-2.1-SNAPSHOT.jar. Where should I find the complete flink-connector-oracle-cdc-2.1-SNAPSHOT.jar

**Reply by leonardBang** [reply]
please see the tips, the oracle cdc is not released yet, you need to compile master by yourself
